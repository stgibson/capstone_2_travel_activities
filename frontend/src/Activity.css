.Activity {
  margin-bottom: 10px;
}

.Activity-link {
  margin-right: 10px;
}