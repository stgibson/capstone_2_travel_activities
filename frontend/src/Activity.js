import React from "react";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";

const Activity = ({ activity, btnText, btnCallback }) => {
  return (
    <div id={ activity.id }>
      <Link to={ `/activities/${activity.id}` }>{ activity.name }</Link>
      {
        btnText && (
          <Button variant="primary" onClick={ btnCallback }>
            { btnText }
          </Button>
        )
      }
    </div>
  );
};

export default Activity;