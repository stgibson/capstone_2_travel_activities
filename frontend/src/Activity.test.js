import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import Activity from "./Activity";

const activity = { id: 1, name: "Test Activity" };
const btnCallback = jest.fn();

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <Activity
        activity={ activity }
        btnText="Button Text"
        btnCallback={ btnCallback }
      />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Activity
        activity={ activity }
        btnText="Button Text"
        btnCallback={ btnCallback }
      />
    </MemoryRouter>
  );
  
  expect(asFragment()).toMatchSnapshot();
});