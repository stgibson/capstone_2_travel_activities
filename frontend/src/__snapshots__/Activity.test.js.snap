// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Activity"
    id="1"
  >
    <a
      class="Activity-link"
      href="/activities/1"
    >
      Test Activity
    </a>
    <button
      class="btn btn-primary btn-sm"
      type="button"
    >
      Button Text
    </button>
  </div>
</DocumentFragment>
`;
